### ISSUE TITLE
sonar-tools hangs when exporting findings without main branch

### ISSUE THREAD

**Issue by vvangestel** [okorach/sonar-tools#1196]
### Description
I ran into an issue with the latest release (3.2) where the program would throw an error and then infinitely hang. The stack trace implies that it is possible for the `main_branch()` of a project to be undefined (`None`), however the code doesn't seem to take this into account when initializing a finding.

### Stack trace
```
2024-07-15 15:19:15,196 | sonar-findings-export | INFO    | MainThread      | Starting finding writer thread 'findingWriter'
Exception in thread findingSearch0:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/cli/findings_export.py", line 317, in __get_component_findings
    findings_list = component.get_issues(filters=new_params)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/projects.py", line 694, in get_issues
    return super().get_issues(filters)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/components.py", line 134, in get_issues
    issue_list = search_all(endpoint=self.endpoint, params=params)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/issues.py", line 695, in search_all
    issue_list = search(endpoint=endpoint, params=params)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/issues.py", line 775, in search
    issue_list[i["key"]] = get_object(endpoint=endpoint, key=i["key"], data=i)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/issues.py", line 838, in get_object
    _ = Issue(endpoint=endpoint, key=key, data=data, from_export=from_export)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/issues.py", line 130, in __init__
    super().__init__(endpoint=endpoint, key=key, data=data, from_export=from_export)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/findings.py", line 123, in __init__
    self._load(data, from_export)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/findings.py", line 130, in _load
    self._load_from_search(data)
  File "/var/lib/jenkins/.local/lib/python3.9/site-packages/sonar/findings.py", line 165, in _load_from_search
    self.branch = projects.Project.get_object(self.endpoint, self.projectKey).main_branch().name
AttributeError: 'NoneType' object has no attribute 'name'
```

### Environment
* Sonar server version: `Community EditionVersion 10.2.1 (build 78527)`
* sonar-tools version: `3.2`

**Comment by FWest98** [comment]
For me, a very similar thing happens when exporting findings for a PR, instead of only branches. While I get similar behaviour, I get a different stacktrace:
```
> sonar-findings-export --format sarif -k foobar -u https://sonarcloud.io/ -t token -o foo -p ID -b develop

2024-07-17 02:56:21,906 | sonar-findings-export | INFO    | MainThread      | Exporting findings for 1 projects with params {'projectKeys': ['foobar'], 'format': 'sarif', 'csvSeparator': ',', 'threads': 8, 'compType': 'projects', 'pullRequests': ['ID'], 'useFindings': False, 'sarifNoCustomProperties': False, 'withURL': False, 'datesWithoutTime': False, 'org': 'foo', 'http_timeout': 10, 'file': None}
Exception in thread findingSearch0:
Traceback (most recent call last):
2024-07-17 02:56:21,906 | sonar-findings-export | INFO    | MainThread      | Starting finding writer thread 'findingWriter'
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/Floris/.local/lib/python3.10/site-packages/cli/findings_export.py", line 317, in __get_component_findings
    findings_list = component.get_issues(filters=new_params)
  File "/home/Floris/.local/lib/python3.10/site-packages/sonar/projects.py", line 692, in get_issues
    branches_or_prs = self.get_branches_and_prs(filters)
  File "/home/Floris/.local/lib/python3.10/site-packages/sonar/projects.py", line 627, in get_branches_and_prs
    objects = {**objects, **pr}
TypeError: 'list' object is not a mapping
^CTraceback (most recent call last):
  File "/home/Floris/.local/bin/sonar-findings-export", line 8, in <module>
    sys.exit(main())
  File "/home/Floris/.local/lib/python3.10/site-packages/cli/findings_export.py", line 414, in main
    store_findings(components_list, params=params)
  File "/home/Floris/.local/lib/python3.10/site-packages/cli/findings_export.py", line 363, in store_findings
    my_queue.join()
  File "/usr/lib/python3.10/queue.py", line 90, in join
    self.all_tasks_done.wait()
  File "/usr/lib/python3.10/threading.py", line 320, in wait
    waiter.acquire()
KeyboardInterrupt
```

**Comment by okorach** [comment]
Hi @vvangestel ,
Thanks for reporting this. I will have a look and may release a patch

**Comment by okorach** [comment]
Hello @vvangestel and @FWest98,

Your problems should be solved with the just released version 3.2.1

Let me know if you still have problems (hopefully not)
